DuggaSys/profileServices/changeProfileValues.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+date_default_timezone_set("Europe/Stockholm");
+
+include_once "../Shared/basic.php";
+include_once "../Shared/sessions.php";
+include_once "database.php"; // Include database connection file
+
+function updateSecurityQuestion($userid, $question, $answer) {
+    pdoConnect(); // Connect to the database
+    $querystring = "UPDATE user SET securityquestion = :question, securityquestionanswer = :answer WHERE uid = :userid";
+    $stmt = $pdo->prepare($querystring);
+    $stmt->bindParam(':question', $question);
+    $stmt->bindParam(':answer', $answer);
+    $stmt->bindParam(':userid', $userid);
+    return $stmt->execute();
+}
+
+function updatePassword($userid, $newPassword) {
+    pdoConnect(); // Connect to the database
+    $hashedPassword = standardPasswordHash($newPassword);
+    $querystring = "UPDATE user SET password = :password WHERE uid = :userid";
+    $stmt = $pdo->prepare($querystring);
+    $stmt->bindParam(':password', $hashedPassword);
+    $stmt->bindParam(':userid', $userid);
+    return $stmt->execute();
+}
+
+function getUserInfo($userid) {
+    pdoConnect();
+    $userInfo = array();
+
+    $querystring = "SELECT * FROM selectFromTableUser WHERE uid = :userid";
+    $stmt = $pdo->prepare($querystring);
+    $stmt->bindParam(':userid', $userid);
+
+    if ($stmt->execute()) {
+        $userInfo = $stmt->fetch(PDO::FETCH_ASSOC);
+    } else {
+        $error = $stmt->errorInfo();
+        $userInfo['error'] = "Error fetching user information: " . $error[2];
+    }
+
+    return $userInfo;
+}
+
+function isSuperUser($userid) {
+    pdoConnect();
+    $isSuperUser = false;
+
+    $querystring = "SELECT * FROM selectFromTableUser WHERE uid = :userid AND super_user = 1";
+    $stmt = $pdo->prepare($querystring);
+    $stmt->bindParam(':userid', $userid);
+
+    if ($stmt->execute()) {
+        if ($stmt->rowCount() > 0) {
+            $isSuperUser = true;
+        }
+    } else {
+        $error = $stmt->errorInfo();
+        $isSuperUser = "Error checking super user status: " . $error[2];
+    }
+
+    return $isSuperUser;
+}
+
+function logServiceEvent($log_uuid, $event_type, $file, $userid, $info) {
```

Comment:
There is a function in basics.php that is named logServiceEvent, that looks a bit like this, is this needed or are you suppose to use the existing function that you imported from basics.php?